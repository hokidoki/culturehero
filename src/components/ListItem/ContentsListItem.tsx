import type ContentListItemProps from './ContentsListItem.types';
import ContentListItemStyles from './ContentsListItem.styles';
import Link from 'next/link';

export default function ContentsListItem({
  id,
  title,
  summary,
}: ContentListItemProps) {
  return (
    <Link href={`/contents/${id}`} style={{ display: 'contents' }}>
      <ContentListItemStyles.Layout>
        <ContentListItemStyles.Title>{title}</ContentListItemStyles.Title>
        <ContentListItemStyles.Summary>{summary}</ContentListItemStyles.Summary>
      </ContentListItemStyles.Layout>
    </Link>
  );
}
