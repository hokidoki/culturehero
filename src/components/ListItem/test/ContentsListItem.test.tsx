import { render } from '@render';
import ContentsListItem from '../ContentsListItem';

describe('ContentsListItem', () => {
  it('match snapshot', () => {
    const props = {
      id: '0',
      title: 'TITLE',
      summary: 'SUMMARY',
      createdAt: '22-02-22',
    };
    const utils = render(<ContentsListItem {...props} />);
    expect(utils.container).toMatchSnapshot();
  });
});
