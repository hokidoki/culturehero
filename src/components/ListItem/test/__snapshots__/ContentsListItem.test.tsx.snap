// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommentEditor match snapshot 1`] = `
.c0 {
  display: flex;
  flex-direction: column;
  justify-content: flex-start;
  align-items: flex-start;
  border: 1px solid rgb(46,48,51);
  border-radius: 15px;
  transition: background-color 0.3s;
  width: 100%;
  height: 100px;
  padding: 10px;
}

.c0:hover {
  background-color: rgb(33,33,36);
  cursor: pointer;
}

.c1 {
  color: #9aa0a6;
  font-size: 1.25rem;
}

.c2 {
  color: #bdc1c6;
  font-size: 1.25rem;
}

<div>
  <a
    href="/contents/0"
    style="display: contents;"
  >
    <div
      class="c0"
    >
      <span
        class="c1"
      >
        TITLE
      </span>
      <p
        class="c2"
      >
        SUMMARY
      </p>
    </div>
  </a>
</div>
`;
